autoload/ale/job.vim

```diff
@@ -188,11 +194,11 @@ function! ale#job#PrepareCommand(buffer, command) abort
         return 'cmd /s/c "' . l:command . '"'
     endif
 
-    if &shell =~? 'fish$\|pwsh$'
-        return ['/bin/sh', '-c', l:command]
+    if g:ale_shell =~? 'fish$\|pwsh$'
+        return ['/bin/sh', g:ale_shell_arguments, l:command]
```

Comment:
please revert this to be hardcoded `'-c'` to avoid conflicts with non-sh-compatible shells; because `sh` is standardized we know we can pass `-c` to it.